lib/src/models/parsed_credential.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ledger_cardano/src/utils/validation_exception.dart';
+
+part 'parsed_credential.freezed.dart';
+
+@freezed
+class ParsedCredential with _$ParsedCredential {
+  ParsedCredential._();
+
+  factory ParsedCredential.keyPath({
+    required List<int> path,
+  }) = CredentialKeyPath;
+
+  factory ParsedCredential.keyHash({
+    required String keyHashHex,
+  }) {
+    if (keyHashHex.length != 56) {
```

Comment:
No magic numbers please. Put this `56` value in some constant (either inside this file as a private constant, or in another `constants` file if it's used elsewhere too.